packages/snap-store-mobx/src/Search/Stores/SearchResultStore.ts

```diff
@@ -237,6 +237,41 @@ export class Variants {
 					selection.select(firstAvailableOption.value, true);
 				}
 			});
+		} else {
+			// select first available for each selection
+			this.selections.forEach((selection, idx) => {
+				// filter by first available, then by preselected option preference
+				//make all options available for first selection.
+				const availableOptions = selection.values.filter((value) => (idx == 0 ? true : value.available));
+				const preferedOptions = options[selection.field as keyof typeof options];
+				let preferencedOption = availableOptions[0];
+
+				// if theres a preference for that field
+				if (preferedOptions) {
+					const checkIfAvailable = (preference: string) => {
+						//see if that option is in the available options
+						const availablePreferedOptions = availableOptions.find((value) => value.value.toLowerCase() == preference.toLowerCase());
+
+						//use it
+						if (availablePreferedOptions) {
+							preferencedOption = availablePreferedOptions;
+						}
+					};
+
+					if (typeof preferedOptions == 'object') {
```

Comment:
Preferred `Array.isArray()` if that is what we are expecting.

Reply:
Logic could be simplified if we just always convert to array, or make it require an array.